### PR TITLE
Add Vulkan op Add.

### PR DESCRIPTION
**Pull request by AshkanAliabadi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46900 Add Vulkan op Conv2D.
* **#44017 Add Vulkan op Add.**
* #46481 Add Vulkan tensor copy.

Differential Revision: [D23820826](https://our.internmc.facebook.com/intern/diff/D23820826)

### CODE REVIEW

File: aten/src/ATen/native/vulkan/ops/Add.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+Tensor add_scalar(
+    const Tensor& self_,
```
Comment (IvanKobzarev) [inline aten/src/ATen/native/vulkan/ops/Add.cpp:11]: Nit: usually I saw in pytorch code base that member fields are named with underscore at the end

File: aten/src/ATen/native/vulkan/ops/Add.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          v_self.image(command_buffer),
+          // Read-only access is implied on const tensors and triggers an async
+          // synchronization if necessary.
+          v_other.image(command_buffer),
+          // Object lifetime is managed by the resource pool.
+          // It is OK not to keep track of the handle.
+          context->resource().pool.uniform(block).object);
+    }
+    else {
+      // TODO: Convert all to buffer and dispatch a buffer-only kernel.
```
Comment (IvanKobzarev) [inline aten/src/ATen/native/vulkan/ops/Add.cpp:59]: Hm. Do you try to avoid on demand on conversion to image view to have them only explicitly done?

**Comment by AshkanAliabadi** [comment]
tensor.has_image means can have image or tensor can be represented as an
image. If false, tensor cannot be represented as an image.

File: aten/src/ATen/native/vulkan/ops/Add.cpp (hunk 1 above)
Comment (AshkanAliabadi) [reply aten/src/ATen/native/vulkan/ops/Add.cpp:11]: No that's a good idea.  I'll follow up with the same convention.  I'll use self_arg, etc.  I am not really sue whether we use camelCase or snake_case in PyTorch.  It seems to be a mix.  I can find instances of both even under native.

**Review by IvanKobzarev — APPROVED** [review]

**Pull request closed without merging**